src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable react/jsx-props-no-multi-spaces */
+import _ from 'underscore';
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {FlatList} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import MenuItem from '../../../components/MenuItem';
+import compose from '../../../libs/compose';
+import withLocalize from '../../../components/withLocalize';
+import ONYXKEYS from '../../../ONYXKEYS';
+import {
+    Bank,
+    CreditCard,
+    PayPal,
+    Plus,
+} from '../../../components/Icon/Expensicons';
+
+const propTypes = {
+    // What to do when a menu item is pressed
+    onPress: PropTypes.func.isRequired,
+
+    // Users paypal.me username if they have one
+    payPalMeUsername: PropTypes.string,
+
+    // Array of bank account objects
+    bankAccountList: PropTypes.arrayOf(PropTypes.shape({
+        addressName: PropTypes.string,
+        accountNubmer: PropTypes.string,
+    })),
+
+    // Array of card objects
+    cardList: PropTypes.arrayOf(PropTypes.shape({
+        cardName: PropTypes.string,
+        accountNumber: PropTypes.string,
+    })),
+};
+
+const defaultProps = {
+    payPalMeUsername: '',
+    bankAccountList: [],
+    cardList: [],
+};
+
+class PaymentMethodList extends Component {
+    constructor(props) {
+        super(props);
+
+        this.renderItem = this.renderItem.bind(this);
+    }
+
+    /**
+     * Render item method wraps the prop renderItem to render in a
+     * View component so we can attach an onLayout handler and
+     * measure it when it renders.
+     *
+     * @param {Object} params
+     * @param {Object} params.item
+     * @param {Number} params.index
+     *
+     * @return {React.Component}
+     */
+    renderItem({item, index}) {
+        return (
+            <MenuItem
+                onPress={item.onPress}
+                title={item.title}
+                description={item.description}
+                icon={item.icon}
+                key={`paymentMethod-${index}`}
+            />
+        );
+    }
+
+    render() {
+        const combinedPaymentMethods = [];
```

Comment:
NAB because it's probably fine to leave this in `render`, but you could create a function like `buildPaymentMethodMenuItems` that does lines 75-100 and saves it to state. Then call that function in the `constructor` and again in `componentDidUpdate` if the props change. Not sure if it really makes any difference, but I _think_ it might be more performant to do it that way vs redeclaring the array with every render. Up to you if you want to make that change.